lib/runners/processor/fxcop.rb

```diff
@@ -0,0 +1,119 @@
+module Runners
+  # A processor class for FxCop Analyzers
+  # https://docs.microsoft.com/en-us/visualstudio/code-quality/install-fxcop-analyzers
+  # This is a beta release. We support only repositories that has .csproj file directly under analysis root directory.
+  class Processor::FxCop < Processor
+
+    #parameters
+    FXCOP_ANALYZER_VERSION = '2.9.8'.freeze
+    ANALYSIS_LOGFILE_PATH = '/tmp/sider_analysis_out.json'.freeze
+    RULE_ID_PATTERN = /CA[0-9]+/.freeze
+
+    def self.ci_config_section_name
+      'fxcop'
+    end
+
+    def analyzer_name
+      'fxcop'
+    end
+
+    def analyzer_bin
+      'dotnet'
+    end
+
+    def analyzer_version
+      FXCOP_ANALYZER_VERSION
+    end
+
+    def setup
+      # restore project (install dependencies)
+      cmdline_install_dependency = %w(dotnet restore)
+      _, _, status = capture3(*cmdline_install_dependency)
+      unless status == 0
+        msg = <<~EOS
+          Failed to restore .NET Core Project. You have to put project file (.csproj) under the analysis root. Please check project structure and Sider config file (sider.yml).
+          Note: We only support a project managed with MSBuild. No support for other build systems. (e.g. Nuke, Cake)
+        EOS
+        return Results::Failure.new(guid: guid, message: msg, analyzer: analyzer)
+      end
+
+      # install FxCop Analyzers from nuget Repository
+      cmdline_add_analyzer = ['dotnet', 'add', 'package', 'Microsoft.CodeAnalysis.FxCopAnalyzers', '--version', FXCOP_ANALYZER_VERSION]
+      capture3!(*cmdline_add_analyzer)
+
+      yield
+    end
+
+    def analyze(changes)
+      # run 'dotnet build' with static analysis module
+      cmdline_run_analyzer = ['dotnet', 'build', '--no-incremental', "-property:errorlog=#{ANALYSIS_LOGFILE_PATH}"]
+      capture3!(*cmdline_run_analyzer)
+
+      issues = parse_result_file(File.open("#{ANALYSIS_LOGFILE_PATH}"))
```

Comment:
[suggestion] `#read` ensures closing, so it is better:

```suggestion
      issues = parse_result_file(File.read(ANALYSIS_LOGFILE_PATH))
```

See https://ruby-doc.org/core-2.7.0/IO.html#method-c-read

Reply:
I modified how to manage output file.
https://github.com/sider/runners/pull/731/commits/9de36274546eb4eabbb0232d0d168e583b3e1c04